File: Jack.Business/Colaborador.vb
Public Class Colaborador
    Implements ICrud(Of Model.Colaborador, Integer)

    Public Sub New()

    End Sub

#Region "Outros Métodos"

    Public Function ListaQuantidadeSacolasPorColaborador(intAno As Integer) As List(Of Model.Colaborador)

        Dim oDados As Data.Colaborador = Nothing
        Dim lstRetorno As IList(Of Model.Colaborador) = Nothing

        Try
            oDados = New Data.Colaborador
            lstRetorno = oDados.ListaQuantidadeSacolasPorColaborador(intAno)
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

#End Region

#Region "Crud"
    Public Function Delete(oTipo As Model.Colaborador) As Boolean Implements ICrud(Of Model.Colaborador, Integer).Delete
        Dim oDados As Data.Colaborador = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Colaborador
            blnRetorno = oDados.Delete(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function

    Public Function Find(Identifier As Integer) As Model.Colaborador Implements ICrud(Of Model.Colaborador, Integer).Find

        Dim oDados As Data.Colaborador = Nothing
        Dim oRetorno As Model.Colaborador = Nothing

        Try
            oDados = New Data.Colaborador
            oRetorno = oDados.Find(Identifier)
        Catch ex As Exception
            oRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return oRetorno

    End Function

    Public Function Insert(oTipo As Model.Colaborador) As Boolean Implements ICrud(Of Model.Colaborador, Integer).Insert
        Dim oDados As Data.Colaborador = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Colaborador
            blnRetorno = oDados.Insert(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Public Function LoadAll() As IList(Of Model.Colaborador) Implements ICrud(Of Model.Colaborador, Integer).LoadAll
        Dim oDados As Data.Colaborador = Nothing
        Dim lstRetorno As IList(Of Model.Colaborador) = Nothing

        Try
            oDados = New Data.Colaborador
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Update(oTipo As Model.Colaborador) As Boolean Implements ICrud(Of Model.Colaborador, Integer).Update
        Dim oDados As Data.Colaborador = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Colaborador
            blnRetorno = oDados.Update(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function
#End Region

End Class

File: Jack.Business/Colaborador.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Business
{
    public class Colaborador : ICrud<Model.Colaborador, int>
    {


        public Colaborador()
        {
        }

        #region "Outros Métodos"

        public IList<Model.Colaborador> ListaQuantidadeSacolasPorColaborador(int intAno)
        {

            Data.Colaborador oDados = null;
            IList<Model.Colaborador> lstRetorno = null;

            try
            {
                oDados = new Data.Colaborador();
                lstRetorno = oDados.ListaQuantidadeSacolasPorColaborador(intAno);
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        #endregion

        #region "Crud"
        public bool Delete(Model.Colaborador oTipo)
        {
            Data.Colaborador oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Colaborador();
                blnRetorno = oDados.Delete(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }

        public Model.Colaborador Find(int Identifier)
        {

            Data.Colaborador oDados = null;
            Model.Colaborador oRetorno = null;

            try
            {
                oDados = new Data.Colaborador();
                oRetorno = oDados.Find(Identifier);
            }
            catch (Exception ex)
            {
                oRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return oRetorno;

        }

        public bool Insert(Model.Colaborador oTipo)
        {
            Data.Colaborador oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Colaborador();
                blnRetorno = oDados.Insert(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public IList<Model.Colaborador> LoadAll()
        {
            Data.Colaborador oDados = null;
            IList<Model.Colaborador> lstRetorno = null;

            try
            {
                oDados = new Data.Colaborador();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Update(Model.Colaborador oTipo)
        {
            Data.Colaborador oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Colaborador();
                blnRetorno = oDados.Update(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }
        #endregion

    }
}
